pkg/controller/infrastructure/infraflow/persistent_state.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infraflow
+
+import (
+	"fmt"
+
+	"github.com/gardener/gardener-extension-provider-aws/pkg/controller/infrastructure/infraflow/shared"
+	"k8s.io/apimachinery/pkg/util/json"
+)
+
+const (
+	// PersistentStateVersion is the current version used for persisting the state.
+	PersistentStateVersion = "1.0"
```

Comment:
maybe `v1`

Reply:
using TypeObject with kind and apiVersion instead. I'm using now same the apiVersion as the other apis types: `aws.provider.extensions.gardener.cloud/v1alpha1`